pkg/cache/v3/resource_test.go

```diff
@@ -129,8 +121,8 @@ func TestGetResourceReferences(t *testing.T) {
 			out: map[rsrc.Type]map[string]bool{rsrc.EndpointType: {"test": true}},
 		},
 		{
-			in:  resource.MakeScopedRouteHTTPListener(resource.Xds, listenerName, 80, scopedRouteName),
-			out: map[rsrc.Type]map[string]bool{rsrc.ScopedRouteType: {scopedRouteName: true}},
+			in:  resource.MakeScopedRouteHTTPListener(resource.Xds, listenerName, 80),
```

Comment:
Should we add a test case to check the embedded routes case? (I.e. MakeScopedRouteHTTPListenerForRoutes)

Reply:
Yeh we might as well. Added this.